### DISCUSSION TITLE
is it posbile to fit Model into camera?

### DISCUSSION THREAD

**Discussion by goldenratio** [jnsmalm/pixi3d#55] [Q&A]
GLTF models are of different sizes, is it possible to show whole model within the camera.
something like this? https://github.com/mrdoob/three.js/issues/6784

**Comment by goldenratio** [comment]
```
camera.distance = modelHeight / 2 / Math.tan(Math.PI * camera.camera.fieldOfView / 360);
```
not sure how to find model height

**Reply by jnsmalm** [reply]
To find the model height you have to loop through all model meshes and find the min/max positions in the geometry.

```
for (let mesh of model.meshes) {
  for (let j = 0; j < mesh.geometry.positions.buffer.length; j += 3) {
    const x = mesh.geometry.positions.buffer[j];
    const y = mesh.geometry.positions.buffer[j + 1];
    const z = mesh.geometry.positions.buffer[j + 2];
}
```

**Reply by jnsmalm** [reply]
Planning to make this a bit easier in the future
